reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -75,6 +98,50 @@ public static Result canSend(Context context) {
         return range.contains(new Time()) ? Result.BLACKOUT : Result.OK;
     }
 
+    private static boolean canSendOnWwanDischarging(Context context, Preferences preferences, BlackoutTime blackoutTime) throws ParseException {
+        return !blackoutTime.canSendOnWwan(new GregorianCalendar()) || !preferences.canSendOffCharge() && !power.isCharging(context);
```

Comment:
Could you please use parentheses to make explicit what is the priority of the || and the && in this expression (a || b) && c is not equal to a || (b && c).